### ISSUE TITLE
NumberBox: screen reader reads "new line" when mouse clicks arrows

### ISSUE THREAD

**Issue by carlos-zamora** [microsoft/microsoft-ui-xaml#10945]
### Describe the bug

With Narrator enabled, clicking on the arrow buttons for the `NumberBox` control results in the screen reader reading out "new line". The announcement should instead reflect the updated value. Issue repros in WinUI 2 Gallery > [NumberBox page](winui2gallery://item/NumberBox)

### Why is this important?

Screen readers are useful for limited-visibility users. A scenario like this is important because this impacts users who use a screen reader and a mouse.

### Steps to reproduce the bug

1. In WinUI 2 Gallery, go to [NumberBox page](winui2gallery://item/NumberBox)
2. Turn on the narrator using 'CTRL+Win+Enter'
3. Change the value by up and down arrow buttons

### Actual behavior

"New line" read out

### Expected behavior

The updated value should be announced

### Screenshots

https://github.com/user-attachments/assets/2996eca8-6829-45f1-bc49-484f907a5bab

### NuGet package version

None

### Additional context

Originally filed in https://github.com/microsoft/terminal/issues/19693

**Comment by niels9001** [comment]
~~I'm not able to repro this in WinUI3 - it announces the correct values.~~
